evo1/atb/entity.py

```diff
@@ -0,0 +1,173 @@
+import logging
+from enum import Enum, auto
+
+from evo1.memory.atb import BattleEntity
+
+logger = logging.getLogger(__name__)
+
+
+class ATBEntityID(Enum):
+    CLINK = auto()
+    KAERIS = auto()
+    # Bosses
+    KEFKAS_GHOST = auto()
+    ZEPHYROS = auto()
+    # Overworld 2D
+    SLIME = auto()
+    EMUK = auto()
+    # Cavern
+    SCAVEN = auto()
+    KOBRA = auto()
+    TORK = auto()
+    # Overworld 3D
+    ZOOMBA = auto()
+    APIDYA = auto()
+    ATUIN = auto()
+
+
+class ATBEntity:
+    def __init__(self, kind: ATBEntityID, turn_gauge: float) -> None:
+        self.kind = kind
+        self.turn_gauge = turn_gauge
+
+
+class ATBEntityStats:
+    def __init__(
+        self,
+        max_hp: int,
+        cur_hp: int,
+        attack: int,
+        defense: int,
+        magic: int,
+        evade: int,
+    ) -> None:
+        self.cur_hp = cur_hp
+        self.max_hp = max_hp
+        self.attack = attack
+        self.defense = defense
+        self.magic = magic
+        self.evade = evade
+
+
+def atb_stats_from_id(entity_id: ATBEntityID, level: int = 0) -> ATBEntityStats:
+    match entity_id:
+        case ATBEntityID.CLINK:
+            max_hp = 100
+            if level >= 3:
```

Comment:
Note, the level conditionals here are not technically correct/generic, but they sufficiently cover the entire Any% route.